File: transform_citations.py
import re

file_path = 'frontend/src/utils/citations.ts'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Replace pattern: "text - author", with "text (author)",
# Match: quote, text, " - ", author, closing quote and comma
pattern = r'(".*?)\s+-\s+(.*?)(",)'

def replace_citation(match):
    quote_part = match.group(1)  # Everything before " - "
    author_part = match.group(2)  # The author name
    closing = match.group(3)      # The ", at the end
    return f'{quote_part} ({author_part}){closing}'

content = re.sub(pattern, replace_citation, content)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print('Citations transformed successfully!')

File: transform-citations.js
const fs = require("fs");
const path = require("path");

const filePath = path.join(
  __dirname,
  "frontend",
  "src",
  "utils",
  "citations.ts"
);
let content = fs.readFileSync(filePath, "utf8");

// Process line by line using simple string manipulation
const lines = content.split("\n");
const transformedLines = lines.map((line) => {
  const trimmed = line.trim();

  // Check if this looks like a citation line: starts with " and contains " - "
  if (
    trimmed.startsWith('"') &&
    trimmed.includes(" - ") &&
    trimmed.endsWith('",')
  ) {
    // Find the last " - " before the closing quote
    const lastDashIndex = trimmed.lastIndexOf(" - ");
    if (lastDashIndex > 0) {
      const quotePart = trimmed.substring(0, lastDashIndex);
      const authorPart = trimmed.substring(lastDashIndex + 3); // +3 to skip " - "

      // Remove the closing quote and comma from author
      if (authorPart.endsWith('",')) {
        const author = authorPart.slice(0, -2); // Remove '",'
        // Get the leading whitespace from original line
        const leadingWhitespace = line.match(/^\s*/)[0];
        return leadingWhitespace + quotePart + " (" + author + ')",';
      }
    }
  }
  return line;
});

fs.writeFileSync(filePath, transformedLines.join("\n"));
console.log(
  `Processed ${lines.length} lines. Citations transformed successfully!`
);
